### ISSUE TITLE
Missing attention_mask in hidden_states calculation causes blurry images in batch inference

### ISSUE THREAD

**Issue by qqzy** [JiuhaiChen/BLIP3o#83]
**Describe the bug**
When performing batch inference, there is a bug in the calculation of hidden_states because the attention mask is not passed to the model. As a result, only the image corresponding to the shortest prompt in the batch is generated correctly (since it does not require padding tokens), while the rest of the images appear blurry.

**Root Cause**
The issue is due to missing attention mask input when calling the model to compute hidden_states. This causes improper handling of padding tokens for longer prompts.

**Solution**
I fixed the issue by adding the attention mask when computing hidden_states:
```
# https://github.com/JiuhaiChen/BLIP3o/blob/BLIP3o-NEXT/blip3o/model/language_model/blip3o_qwen_inference.py

gen_attention_mask = torch.cat([attention_mask, torch.ones(attention_mask.size(0), max_new_tokens, dtype=attention_mask.dtype, device=attention_mask.device)],dim=1)
with torch.no_grad():
    outs = self.model(
        input_ids = gen_ids, 
        attention_mask = gen_attention_mask,
        output_hidden_states = True,
        return_dict = True,
    )
```

After adding the attention mask, the problem was resolved and all prompts generated clear images.

Would you like me to submit a pull request with the fix?

**Comment by JiuhaiChen** [comment]
That would be great if you submit a PR to support the batch inference.

**Comment by qqzy** [comment]
Okay, I'll submit a PR.
I have another question. I noticed that in the RL training code you provided, batch inference is also used during the sampling process. Why weren't you affected by this bug?
https://github.com/JiuhaiChen/BLIP3o/blob/ad495eb94097944654e15b67f59ade8477a3c02e/trl/trl/trainer/grpo_trainer.py#L1643-L1655

**Comment by qqzy** [comment]
> Okay, I'll submit a PR. I have another question. I noticed that in the RL training code you provided, batch inference is also used during the sampling process. Why weren't you affected by this bug?
> 
> [BLIP3o/trl/trl/trainer/grpo_trainer.py](https://github.com/JiuhaiChen/BLIP3o/blob/ad495eb94097944654e15b67f59ade8477a3c02e/trl/trl/trainer/grpo_trainer.py#L1643-L1655)
> 
> Lines 1643 to 1655 in [ad495eb](/JiuhaiChen/BLIP3o/commit/ad495eb94097944654e15b67f59ade8477a3c02e)


One possible reason is that the author set num_generations == per_device_train_batch_size in the training parameters. This means that in a single process, all prompts in a batch are identical; therefore, no padding tokens exist.

**Comment by JiuhaiChen** [comment]
Hi, thanks for submitting PR. In RL, we didn't use the batch inference.
